src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveInputStream.java

```diff
@@ -931,6 +942,19 @@ public int read(final byte[] buffer, final int offset, final int length) throws
         } else if (current.entry.getMethod() == ZipMethod.UNSHRINKING.getCode() || current.entry.getMethod() == ZipMethod.IMPLODING.getCode()
                 || current.entry.getMethod() == ZipMethod.ENHANCED_DEFLATED.getCode() || current.entry.getMethod() == ZipMethod.BZIP2.getCode()) {
             read = current.inputStream.read(buffer, offset, length);
+        } else if (current.entry.getMethod() == ZipMethod.ZSTD.getCode() || current.entry.getMethod() == ZipMethod.ZSTD_DEPRECATED.getCode()) {
```

Comment:
This can be done the same as deflate, removing the need for the one-off instance variable, done in git master.
More importantly, this code does not appear to be tested, as putting breakpoints in the debugger on lines 947, 951, and 954 are never hit when running the new test.